src/test/app/LoanBroker_test.cpp

```diff
@@ -851,12 +851,335 @@ class LoanBroker_test : public beast::unit_test::suite
+        }
+    }
+
+    void
+    testInvalidLoanBrokerCoverClawback()
+    {
+        testcase("Invalid LoanBrokerCoverClawback");
+        using namespace jtx;
+        using namespace loanBroker;
+
+        // preflight
+        {
+            Account const alice{"alice"};
+            Account const issuer{"issuer"};
+            auto const USD = alice["USD"];
+            Env env(*this);
+            env.fund(XRP(100'000), alice);
+            env.close();
+
+            auto jtx = env.jt(coverClawback(alice), amount(USD(100)));
+
+            // holder == account
+            env(jtx, ter(temINVALID));
+
+            // holder == beast::zero
+            STAmount bad(Issue{USD.currency, beast::zero}, 100);
+            jtx.jv[sfAmount] = bad.getJson();
+            jtx.stx = env.ust(jtx);
+            Serializer s;
+            jtx.stx->add(s);
+            auto const jrr = env.rpc("submit", strHex(s.slice()))[jss::result];
+            // fails in doSubmit() on STTx construction
+            BEAST_EXPECT(jrr[jss::error] == "invalidTransaction");
+            BEAST_EXPECT(jrr[jss::error_exception] == "invalid native account");
+        }
+
+        // preclaim
+
+        // Issue:
+        // AllowTrustLineClawback is not set or NoFreeze is set
+        testLoanBroker({}, CoverClawback);
+
+        // MPTIssue:
+        // MPTCanClawback is not set
+        testLoanBroker(
+            [&](Env& env, Account const& issuer, Account const& alice) -> MPT {
+                MPTTester mpt(
+                    {.env = env, .issuer = issuer, .holders = {alice}});
+                return mpt;
+            },
+            CoverClawback);
+        // MPTCanLock is not set
+        testLoanBroker(
+            [&](Env& env, Account const& issuer, Account const& alice) -> MPT {
+                MPTTester mpt(
+                    {.env = env,
+                     .issuer = issuer,
+                     .holders = {alice},
+                     .flags = MPTDEXFlags | tfMPTCanClawback});
+                return mpt;
+            },
+            CoverClawback);
+    }
+
+    void
+    testInvalidLoanBrokerCoverDeposit()
+    {
+        testcase("Invalid LoanBrokerCoverDeposit");
+        using namespace jtx;
+
+        // preclaim:
+        // tecWRONG_ASSET, tecINSUFFICIENT_FUNDS, frozen asset
+        testLoanBroker({}, CoverDeposit);
+    }
+
+    void
+    testInvalidLoanBrokerCoverWithdraw()
+    {
+        testcase("Invalid LoanBrokerCoverWithdraw");
+        using namespace jtx;
+
+        /*
+        preflight: illegal net
+        isLegalNet() check is probably redundant. STAmount parsing
+        should throw an exception on deserialize
+
+        preclaim: tecWRONG_ASSET, tecNO_DST, tecDST_TAG_NEEDED,
+            tecNO_PERMISSION, checkFrozen failure, checkDeepFrozenFailure,
+            second+third tecINSUFFICIENT_FUNDS (can this happen)?
+        doApply: tecPATH_DRY (can it happen, funds already checked?)
```

Comment:
`tecPATH_DRY` is a catch all for any non-tec payment engine failure. Because the engine is large and complex, there are other things that can fail other than available funds.